liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -959,4 +854,102 @@ public void info(String msg) {
+        Plugin libertyPlugin = getLibertyPluging();
+        Xpp3Dom overrides = configuration(element(name("looseApplication"), "true"));
+        return getPluginConfig(libertyPlugin, overrides);
+    }
+
+    private Xpp3Dom getPluginConfig(Plugin plugin, Xpp3Dom overrides) {
+        Xpp3Dom config = (overrides != null) ? overrides : configuration();
+        if (plugin.getConfiguration() == null) {
+            return config;
+        } else {
+            return Xpp3Dom.mergeXpp3Dom(config, (Xpp3Dom)plugin.getConfiguration());
+        }
+    }
+
+    private final ArrayList<String> commonParams = new ArrayList<>(Arrays.asList(
+            "installDirectory", "runtimeArchive", "runtimeArtifact", "libertyRuntimeVersion",
+            "install", "licenseArtifact", "serverName", "userDirectory", "outputDirectory",
+            "runtimeInstallDirectory", "refresh", "skip"
+            ));
+    
+    private final ArrayList<String> commonServerParams = new ArrayList<>(Arrays.asList(
+            "serverXmlFile", "configDirectory", "bootstrapProperties", "bootstrapPropertiesFile",
+            "jvmOptions", "jvmOptionsFile", "serverEnvFile"));
+    private ArrayList<String> createParams = new ArrayList<>(Arrays.asList(
+            "template", "libertySettingsFolder", "noPassword"
+            ));
+    
+    private ArrayList<String> deployParams = new ArrayList<>(Arrays.asList(
+            "appsDirectory", "stripVersion", "deployPackages", "looseApplication", "timeout"
+            ));
+    
+    private ArrayList<String> installFeatureParams = new ArrayList<>(Arrays.asList("features"));    
+
+    private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
+        for (int i=0; i<config.getChildCount(); i++) {
+            if (!goalParams.contains(config.getChild(i).getName().trim())) {
+                config.removeChild(i);
+                stripConfigElements(config, goalParams);
+            }
+        }
+        return config;
+    }
+
+    private void runLibertyMojoCreate() throws MojoExecutionException {
+        createParams.addAll(commonParams);
+        createParams.addAll(commonServerParams);
+        Xpp3Dom config = stripConfigElements(getLibertyPluginConfig(), createParams);
+        log.info("Running liberty:create goal");
+        runLibertyMojo("create", config);
+    }
+
+    private void runLibertyMojoDeploy() throws MojoExecutionException {
+        deployParams.addAll(commonParams);
+        deployParams.addAll(commonServerParams);
+        Xpp3Dom config = stripConfigElements(getLibertyPluginConfig(), deployParams);
+        stripConfigElements(config, deployParams);
+        log.info("Running liberty:deploy goal");
+        runLibertyMojo("deploy", config);
+    }
+
+    private void runLibertyMojoInstallFeature() throws MojoExecutionException {
+        installFeatureParams.addAll(commonParams);
```

Comment:
Same comment as `createParams`.

Reply:
will address them and make them static class variables